src/gpu/intel/jit/ir/tensor.hpp

```diff
@@ -406,22 +375,10 @@ class layout_t {
     int nblocks() const { return (int)blocks().size(); }
 
     const std::vector<layout_block_t> &blocks() const { return blocks_; }
+    std::vector<layout_block_t> &blocks() { return blocks_; }
 
-    dim_t inner_block(const pvar_t &dim, bool skip_outer = true,
-            bool inner_only = true) const {
-        std::vector<dim_t> dim_blocks;
-        for (auto &b : blocks_) {
-            if (b.dim == dim) dim_blocks.push_back(b.block);
-        }
-        dim_t ret = 1;
-        int nblocks = (int)dim_blocks.size();
-        int lo = 0;
-        int hi = skip_outer ? nblocks - 1 : nblocks;
-        if (inner_only) hi = std::min(hi, 1);
-        for (int i = lo; i < hi; i++)
-            ret *= dim_blocks[i];
-        return ret;
-    }
+    const layout_block_t &operator[](size_t idx) const { return blocks_[idx]; }
```

Comment:
I think we should change `nblocks()` to return `size_t` (or vice versa, `int idx` here) - to align.